Linux disk encryption guides (#24115)
fleetdm/fleet: articles/enforce-disk-encryption.md

_Available in Fleet Premium_

> In Fleet, you can enforce disk encryption for your macOS, Windows, Ubuntu Linux, and Fedora Linux hosts.

Reviewer: We can't enforce for Linux

Author: @sharon-fdm wdyt?
```suggestion
In Fleet, you can enforce disk encryption for your macOS and Windows hosts, and verify disk encryption for Ubuntu Linux and Fedora Linux hosts.
```

> Apple calls this [FileVault](https://support.apple.com/en-us/HT204837), Microsoft calls this [BitLocker](https://learn.microsoft.com/en-us/windows/security/operating-system-security/data-protection/bitlocker/), and Linux typically uses [LUKS](https://en.wikipedia.org/wiki/Linux_Unified_Key_Setup) (Linux Unified Key Setup).

When disk encryption is enforced, hosts' disk encryption keys will be stored in Fleet.

> For macOS hosts that automatically enroll, disk encryption is enforced during Setup Assistant. For Windows, disk encryption is enforced on the C: volume (default system/OS drive). On Linux, encryption enforcement involves user interaction to encrypt the device with LUKS.

Reviewer: Reconsider the word `enforce` for Linux?

Author: ```suggestion
For macOS hosts that automatically enroll, disk encryption is enforced during Setup Assistant. For Windows, disk encryption is enforced on the C: volume (default system/OS drive). On Linux, encryption requires user interaction to encrypt the device with LUKS.
```

When disk encryption is enforced, hosts' disk encryption keys will be stored in Fleet.

For macOS hosts that automatically enroll, disk encryption is enforced during Setup Assistant. For Windows, disk encryption is enforced on the C: volume (default system/OS drive). On Linux, encryption enforcement involves user interaction to encrypt the device with LUKS.

## Enforce disk encryption

You can enforce disk encryption using the Fleet UI, Fleet API, or [Fleet's GitOps workflow](https://github.com/fleetdm/fleet-gitops).

#### Fleet UI:

1. In Fleet, head to the **Controls > OS settings > Disk encryption** page.

2. Choose which team you want to enforce disk encryption on by selecting the desired team in the teams dropdown in the upper left corner.

3. Check the box next to **Turn on** and select **Save**.

#### Fleet API: 

API documentation is [here](https://fleetdm.com/docs/rest-api/rest-api#update-disk-encryption-enforcement).

### Disk encryption status

> In the Fleet UI, head to the **Controls > OS settings > Disk encryption** tab. You will see a table that shows the status of disk encryption on your hosts.

Author: ```suggestion
In the Fleet UI, head to the **Controls > OS settings > Disk encryption** tab. You will see a table that shows the status of disk encryption on your hosts. 
```

* Verified: the host turned disk encryption on and sent their key to Fleet. Fleet verified with osquery. See instructions for viewing the disk encryption key [here](#view-disk-encryption-key).

* Verifying: the host acknowledged the MDM command to install the disk encryption profile. Fleet is verifying with osquery and retrieving the disk encryption key.

> It may take up to one hour for Fleet to collect and store the disk encryption keys from all hosts.

* Action required (pending): the end user must take action to turn disk encryption on or reset their disk encryption key.

* Enforcing (pending): the host will receive the MDM command to install the configuration profile when the host comes online.

* Removing enforcement (pending): the host will receive the MDM command to remove the disk encryption profile when the host comes online.

> * Failed: hosts that are failed to enforce disk encryption.

Author: ```suggestion
* Failed: hosts that failed to enforce disk encryption.
```

> It may take up to one hour for Fleet to collect and store the disk encryption keys from all hosts.

> * Action required (pending): the end user must take action to turn disk encryption on or reset their disk encryption key.

Author: ```suggestion
* Action required (pending): the end user must take action to turn disk encryption on or reset their disk encryption key. 
```

* Removing enforcement (pending): the host will receive the MDM command to remove the disk encryption profile when the host comes online.

* Failed: hosts that failed to enforce disk encryption.

You can click each status to view the list of hosts for that status.

## Enforce disk encryption on Linux

To enforce disk encryption on Ubuntu Linux and Fedora Linux devices, Fleet supports Linux Unified Key Setup (LUKS) for encrypting volumes.

1. Share [this step-by-step guide](https://fleetdm.com/learn-more-about/encrypt-linux-device) with end users setting up a work computer running Ubuntu Linux or Fedora Linux.

> Note that full disk encryption can only enabled during operating system setup. If the operating system has already been installed, the end user will be required to re-install the OS to enable disk encryption.

2. Once the user encrypts the disk, Fleet will initiate a key escrow process through Fleet Desktop:
   * Fleet Desktop prompts the user to enter their current encryption passphrase.
   * A new encryption passphrase is generated and added as a LUKS keyslot for the encrypted volume.

>    * The new passphrase is securely stored in Fleet's backend.

Author: ```suggestion
   * The new passphrase is securely stored in Fleet.
```
